### DISCUSSION TITLE
Problems plotting P4 graphs

### DISCUSSION THREAD

**Discussion by emmamclea** [IBAHCM/RPiR#475] [Archive]
Hi, 
I am having trouble plotting the graphs for practical 4 in the demo. When using the function created and using the vector qs (*qs<-c(0:50)*) (ie *p4_diversity_profile(tree.pop, qs)*) and plotting this, I get a straight line of points across my graph instead of the curved graph we want. When checking the output of the function, ie *p4_diversity_profile(tree.pop, 0)* or *p4_diversity_profile(tree.pop, 1)* etc I get the correct numerical output of 299, then 55.65 etc.

I am not sure why this is happening within my graph. I feel it may be something obvious but I've been going in circles trying to figure it out! Any help would be greatly appreciated.
Best wishes,
Emma

**Comment by richardreeve** [comment]
@emmamclea - there's not a huge amount I can say, but it sounds like your profile function is working fine for single values of q, but doesn't work correctly for multiple values of q. I might try to get it to work for:
```R
qs <- c(0, 1)
p4_diversity_profile(tree.pop, qs)
```
And see if you get `299` *and* `55.65` back... if not, then that's the simplest case to fix first.

**Reply by emmamclea** [reply]
Hi @richardreeve,
When I run the code given, I get the output '55.65416 55.65416'. 
When I change the code to qs <- c(0, 1,2,3), I get the output of 14.94385 14.94385 14.94385 14.94385.

**Reply by richardreeve** [reply]
You really need to work this out for yourself, but without checking the results it looks like you may be calculating the diversity of the last number in the `qs` (in your examples `q=1` and then `q=3`) multiple times rather than calculating the diversity of each value of q one at a time in a loop and combining them to produce the correct answer.
